server.go

```diff
@@ -69,7 +70,7 @@ func NewServer(target string) *Server {
 		Port:             3005,
 		ProxyTimeout:     time.Duration(10) * time.Second,
 		PassThrough:      false,
-		AroundRequest:    func(_ http.ResponseWriter, _ *http.Request, callback func()) { callback() },
+		AroundRequest:    func(h http.Handler) http.Handler { return h },
```

Comment:
What are your thoughts on making `AroundRequest` private and exposing an API like:

```go
func (s *Server) Use(handler func(h http.Handler) http.Handler) {
	prevMiddleware := s.AroundRequest

	s.AroundRequest = func(next http.Handler) http.Handler {
		return handler(prevMiddleware(next))
	}
}
```

I believe that would let us chain middlewares like:

```go
server.Use(logMiddleware)
server.Use(RequestIDMiddleware)
server.Use(FilterMiddleware)
```

Reply:
Yeah, I think what we have now allows consumers of viewproxy to use something like https://github.com/justinas/alice directly with the `AroundRequest` signature which seems like a good minimal API.